src/datapoints-graph/charts/chart-realtime.service.ts

```diff
@@ -32,14 +46,35 @@ export class ChartRealtimeService {
     datapointOutOfSyncCallback: (dp: DatapointsGraphKPIDetails) => void,
     timeRangeChangedCallback: (
       timeRange: Pick<DatapointsGraphWidgetConfig, 'dateFrom' | 'dateTo'>
-    ) => void
+    ) => void,
+    alarmOrEventConfig: AlarmOrEvent[]
   ) {
     this.echartsInstance = echartsInstance;
     this.currentTimeRange = {
       dateFrom: new Date(timeRange.dateFrom),
       dateTo: new Date(timeRange.dateTo),
     };
 
+    const activeAlarmsOrEvents = alarmOrEventConfig.filter(
+      (alarmOrEvent) => alarmOrEvent.__active
+    );
+    const uniqueAlarmOrEventTargets = Array.from(
+      new Set(activeAlarmsOrEvents.map((aOrE) => aOrE.__target.id))
+    );
+
+    const allAlarmsAndEvents: Observable<IAlarm | IEvent | number>[] =
```

Comment:
I don't know how  a number could even be returned in our case... It's only included, because the data property in the RealtimeMessage interface has a type "T | number". I cannot reproduce this number value...